From ecfd59c9459f1c89b43dd3518f1cbe72c2534952 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 26 Mar 2018 10:40:33 +0800
Subject: [PATCH 3548/5242] MLK-17908: ARM64: dts: Add power domains for HDMI
 resources

commit  b2de207cc10e1ec2a47b00b389d0cf83d373f753 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add power domain PD_HDMI_PLL_0/1 and PD_HDMI_I2S.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |   29 ++++++++++++++++++++++---
 include/dt-bindings/soc/imx8_pd.h             |    3 +++
 2 files changed, 29 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 585eb52..46bf2b6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -247,11 +247,34 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				pd_hdmi_i2c0: PD_HDMI_I2C_0 {
-					reg = <SC_R_HDMI_I2C_0>;
+				pd_hdmi_pll0: PD_HDMI_PLL_0{
+					reg = <SC_R_HDMI_PLL_0>;
 					#power-domain-cells = <0>;
 					power-domains =<&pd_hdmi>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					pd_hdmi_pll1: PD_HDMI_PLL_1{
+						reg = <SC_R_HDMI_PLL_1>;
+						#power-domain-cells = <0>;
+						power-domains =<&pd_hdmi_pll0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						pd_hdmi_i2c0: PD_HDMI_I2C_0 {
+							reg = <SC_R_HDMI_I2C_0>;
+							#power-domain-cells = <0>;
+							power-domains =<&pd_hdmi_pll1>;
+						};
+
+						pd_hdmi_i2s: PD_HDMI_I2S {
+							reg = <SC_R_HDMI_I2S>;
+							#power-domain-cells = <0>;
+							power-domains =<&pd_hdmi_pll1>;
+						};
+					};
 				};
+
 			};
 
 		};
@@ -1516,7 +1539,7 @@
 						"clk_lpcg", "clk_even","clk_dbl",
 						"clk_vif", "clk_apb_csr","clk_apb_ctrl",
 						"clk_i2s", "clk_i2s_bypass";
-		power-domains = <&pd_hdmi>;
+		power-domains = <&pd_hdmi_i2s>;
 
 		port@0 {
 			reg = <0>;
diff --git a/include/dt-bindings/soc/imx8_pd.h b/include/dt-bindings/soc/imx8_pd.h
index b7afa19..2152f3a 100644
--- a/include/dt-bindings/soc/imx8_pd.h
+++ b/include/dt-bindings/soc/imx8_pd.h
@@ -188,7 +188,10 @@
 #define PD_PARALLEL_CSI_PLL         parallel_csi_pll
 
 #define PD_HDMI                     hdmi_power_domain
+#define PD_HDMI_PLL_0               hdmi_pll0
+#define PD_HDMI_PLL_1               hdmi_pll1
 #define PD_HDMI_I2C_0               hdmi_i2c
+#define PD_HDMI_I2S_0               hdmi_i2s
 #define PD_HDMI_PWM_0               hdmi_pwm
 #define PD_HDMI_GPIO_0              hdmi_gpio
 
-- 
1.7.9.5

